cli/src/org/partiql/shell/ShellHighlighter.kt

```diff
@@ -13,130 +13,114 @@
-     * highlight tokens
-     */
     override fun highlight(reader: LineReader, line: String): AttributedString {
 
-        val hasAddToGlobalEnv = line.toLowerCase().startsWith(ADD_TO_GLOBAL_ENV)
-        val input = if (hasAddToGlobalEnv) {
-            line.substring(ADD_TO_GLOBAL_ENV.length, line.length)
-        } else {
-            line
+        val hasAddToGlobalEnv = line.toLowerCase().startsWith(ADD_TO_GLOBAL_ENV_STR)
+        val input = when (hasAddToGlobalEnv) {
+            true -> line.substring(ADD_TO_GLOBAL_ENV_STR.length, line.length)
+            false -> line
         }
 
         if (input.isBlank()) {
-            return if (hasAddToGlobalEnv) {
-                AttributedString(line, AttributedStyle.DEFAULT.foreground(AttributedStyle.CYAN))
-            } else {
-                AttributedString(line)
+            return when (hasAddToGlobalEnv) {
+                true -> AttributedString(line, AttributedStyle.DEFAULT.foreground(AttributedStyle.CYAN))
+                false -> AttributedString(line)
             }
         }
 
-        // Map Between Line Number and Index
-        val lineIndexesMap = mutableMapOf<Int, Int>()
-        var currentLine = 0
-        lineIndexesMap[currentLine] = -1
-        input.forEachIndexed { index, char ->
-            if (char == '\n') lineIndexesMap[++currentLine] = index
-        }
-
-        // Check for Non-PartiQL Suffixes (REPL Only)
-        val lastNewlineIndex = lineIndexesMap[currentLine]!!
-        val suffixString = input.substring(lastNewlineIndex + 1, input.lastIndex + 1)
-        val lastValidQueryIndex = when (currentLine > 0 && ALLOWED_SUFFIXES.contains(suffixString)) {
-            true -> lastNewlineIndex
+        // Temporarily Remove Allowed Suffix from Input
+        val lastNewlineIndex = input.indexOfLast { c -> c == '\n' }
+        val suffixString = input.substring(lastNewlineIndex + 1)
+        val lastValidQueryIndex = when (lastNewlineIndex > 0 && ALLOWED_SUFFIXES.contains(suffixString)) {
+            true -> lastNewlineIndex + 1
             false -> input.length
         }
-
-        // Get Tokens
-        val tokens: List<Token>
-        try {
-            tokens = lexer.tokenize(input.substring(0, lastValidQueryIndex))
-        } catch (e: Exception) {
-            val error = AttributedString(input, AttributedStyle().foreground(AttributedStyle.RED))
-            return if (hasAddToGlobalEnv) {
-                with(AttributedStringBuilder()) {
-                    append(ADD_TO_GLOBAL_ENV_COMMAND)
-                    append(error)
-                    toAttributedString()
-                }
-            } else {
-                error
-            }
-        }
+        val usableInput = input.substring(0, lastValidQueryIndex)
 
         // Build Token Colors (Last Token is EOF)
+        val stream = getTokenStream(usableInput)
+        stream.fill()
```

Comment:
```suggestion
        val stream = getTokenStream(usableInput).fill()
```

Reply:
As mentioned above, `.fill()` is a void-returning function. But changing to use `.also()`.